README.md

```diff
@@ -2,36 +2,176 @@
-http://localhost:5000
-spag get /
+pip install .
+```
+
+## Basics
+
+```bash
+$ spag post /things --data '{ "id":"pglbutt" }' -e http://localhost:5000 -H Content-Type:application/json
+{
+  "id": "pglbutt"
+}
+
+$ spag get /api/things -e http://localhost:9001
+{
+  "things": [
+    {
+      "id": "pglbutt"
+    }
+  ]
+}
+
+$ spag put /things/pglbutt --data '{ "id":"pglbutt2" }' -e http://localhost:5000 -H Content-Type:application/json
+{
+  "id": "pglbutt2"
+}
+
+$ spag delete /things/pglbutt2 -e http://localhost:5000
+
+$ spag get /things -e http://localhost:5000
+{
+  "things": []
+}
+```
+
+## Environments
+
+Environments allow you to set arbitrary variables to be used in your request. There are special ones:
+
+* __endpoint__ - The HTTP endpoint that will be postfixed with your request resource.
+* __headers__ - Specified with __-H__, they will be used in all your requests
+* __dir__ - The directory your templates are in
+
+```bash
+# The default environment is always there
+$ spag env set endpoint=http://localhost:5000 id=pglbutt -H Content-Type:application/json
+endpoint: http://localhost:5000
+headers:
+  Content-Type: application/json
+
+# Spag applies your environment variables
+$ spag post /things --data '{ "id":"pglbutt" }'
+{
+  "id": "pglbutt"
+}
+
+$ spag get /things/pglbutt
+{
+  "id": "pglbutt"
+}
+
+# You can also create your own environments out of band of spag
+cat << EOF > .spag/environments/pglbutt.yml
+endpoint: http://localhost:5000
+headers:
+  Content-Type: application/json
+  Some-Other-header: text
+  pglbutt: pglbutt
+EOF
+
+$ spag env activate pglbutt
+Environment pglbutt activated
+
+$ spag env show
+endpoint: http://localhost:5000
+headers:
+  Content-Type: application/json
+  Some-Other-header: text
+  pglbutt: pglbutt
+
+```
+
+## Remembers
+```bash
+$ spag post /things --data '{ "id":"pglbutt" }'
+{
+  "id": "pglbutt"
+}
+# Use '@' to get items from the last response body
+$ spag get /things/@id
 {
-    'name': 'spag'
+  "id": "pglbutt"
 }
 ```
 
+## Templates
+
+## History
+
+```bash
+$ spag history
+0: GET http://localhost:5000/things/pglbutt
+1: POST http://localhost:5000/things
+2: POST http://localhost:5000/clear
+3: DELETE http://localhost:5000/things/pglbutt
+4: GET http://localhost:5000/things/pglbutt
+5: POST http://localhost:5000/things
+6: GET http://localhost:5000/things
+7: POST http://localhost:5000/clear
+
+$ spag history show 7
+-------------------- Request ---------------------
+POST http://localhost:5000/clear
+Content-Length: 0
+Accept-Encoding: gzip, deflate
+Content-Type: application/json
+Connection: keep-alive
+Accept: */*
+User-Agent: python-requests/2.6.2 CPython/2.7.5 Darwin/13.1.0
+-------------------- Response --------------------
+Status code 204
+date: Thu, 28 May 2015 01:58:03 GMT
+content-length: 0
+content-type: text/html; charset=utf-8
+server: Werkzeug/0.10.4 Python/2.7.5
+```
+
 ## Tests
 
+It's really easy to run the tests.
 ```bash
 pip install nose
 nosetests tests/
 ```
```

Comment:
Needs to start the test app: `python tests/testapp.py &`. 

Not in this PR, but we should add a makefile for this or something. Or maybe just `run_tests.sh`. So we don't have to manually kill the test app afterwards.